src/main/java/ro/developmentfactory/thedrones/service/DroneServiceImpl.java

```diff
@@ -0,0 +1,54 @@
+package ro.developmentfactory.thedrones.service;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import ro.developmentfactory.thedrones.entity.Drone;
+import ro.developmentfactory.thedrones.repository.DroneRepository;
+
+import java.sql.Timestamp;
+import java.sql.Date;
+import java.util.List;
+import java.util.Objects;
+
+@Service
+public class DroneServiceImpl implements DroneServcie {
+
+    @Autowired
+    private DroneRepository droneRepository;
+    private DroneStatusService droneStatusService;
+
+    //Read operation
+    @Override public List<Drone> fetchDroneList(){
+        return (List<Drone>) droneRepository.findAll();
+    }
+
+    @Override
+    public Drone saveDrone(Drone drone) {
+        Timestamp currentTime = new Timestamp(System.currentTimeMillis());
+        drone.setCreatedAt(currentTime);
+        drone.setUpdatedAt(currentTime);
+        return droneRepository.save(drone);
+    }
+
+    @Override
+    public Drone updateDrone(Drone drone, Long droneID) {
+        Drone droneDB = droneRepository.findById(droneID).get();
+
+        if (Objects.nonNull(drone.getName()) && !"".equalsIgnoreCase(drone.getName())) {
+            droneDB.setName(drone.getName());
+        }
+        if (Objects.nonNull(drone.getCountMove())) {
+            droneDB.setCountMove(drone.getCountMove());
+        }
```

Comment:
let's better write a separate method that will do only the change of the `count_move`. if you do it all here, it will be much harder to support the code and it will be much more error-prone. and from the API we can only update the name

Reply:
done